### PR TITLE
TT-7898 Opentelemetry e2e tests pipeline

### PR DESCRIPTION
**Pull request by tbuchaillot**
## Description

Setting up an Opentelemetry e2e testing pipeline on CI that:

1.  Builds a Tyk docker image based on current code
2.  Setup `OpenTelemetry collector`
3.  Setup the `tracetest` environment
4.  Execute tests over output traces thru tracetest CLI. The tests are defined under ci/tests/tracing/scenarios

You can also run the tests manually with `task test:opentelemetry` or you can execute each step (particularly useful when adding new tests). If you execute `task -l` you will see all the available commands.

Added a readme for detailed components instruction on  `ci/tests/tracing/README.md`

This PR also includes Test cases for:
- gRPC API + unary (tyk_grpcapi_200.yml)
- HTTP API ok response (tyk_test_200.yml)
- HTTP API Upstream error (tyk_test_500.yml)
- HTTP API Auth middleware error (tyk_testauth_401.yml)
- Tyk protocol API proxying to a open API (non-secured) (tyk_tykprotocol_200.yml)
- Tyk protocol API proxying to a secured API (401 error) (tyk_tykprotocol_401.yml)
## Related Issue

https://tyktech.atlassian.net/browse/TT-7898
## Motivation and Context

https://tyktech.atlassian.net/browse/TT-7898
## How This Has Been Tested

CI :D 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5292/merge
Commit:  
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5486805394)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5292/merge
Commit: 8bcb6fb960069703ac6ee4bb2c7082085e6e6eeb 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5486847031)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5292/merge
Commit: 719a8ec2637f7ec0e03646e58a841b5c2ea5bf63 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5487086537)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5292/merge
Commit: 13258e737bf56da07546a5be238ee4b8410cb5f1 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5487202020)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5292/merge
Commit: 74b6407a146e73d2cf73e177d4dee21d6ffc941e 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5487489509)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5292/merge
Commit: bb6e39b9ff9287cccbda8eb5235c056dbb34eae3 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5488561400)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5292/merge
Commit: ad06757078c087fec7a7c956a11b9eb942812b75 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5488577632)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5292/merge
Commit: 4d951a3844b1f00ea6ba8ec7c7524e8d842dc093 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5488714121)

File: Taskfile.yml [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+includes:
+  deps: ./.taskfiles/deps_{{OS}}.yml
+
 tasks:
+  docker:
+    desc: "build Tyk gateway internal/tyk-gateway"
+    cmds:
+      - docker build --platform "linux/amd64" --rm -t internal/tyk-gateway .
+    sources:
+      - gateway/*.go
```
Comment (titpetric) [inline Taskfile.yml:13]: Could be **.go iirc, also need go.mod go.sum

File: Taskfile.yml [hunk 2]
```diff
@@ -25,3 +46,40 @@ tasks:
...
+      - deps:tracetest
+      - docker
+    status:
+      - tracetest version
+    cmds:
+      - docker compose -f ci/tests/tracing/docker-compose.yml up -d
+      - tracetest configure -g --endpoint http://localhost:11633
+
+
+  test:opentelemetry:test:
```
Comment (titpetric) [inline Taskfile.yml:62]: Drop the test prefix and move opentelemetry into .taskfiles/otel/Taskfile to drop the opentelemetry prefix from yaml. Include otel taskfile into root taskfile

File: ci/tests/tracing/post.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+##TODO define create_keys() and exceed_limit() to setup AuthAPI and RateLimitAPI for traces
```
Comment (titpetric) [inline ci/tests/tracing/post.sh:3]: No new todos in code

File: .github/workflows/opentelemetry-e2e.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+jobs:
+  e2e:
+    name: Opentelemetry e2e
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
```
Comment (titpetric) [inline .github/workflows/opentelemetry-e2e.yml:15]: Fetch depth 1? Missing git fetch github.base_ref? Why is the value set to what it is?

**Review by titpetric — APPROVED** [review]

File: Taskfile.yml (hunk 2 above)
Comment (tbuchaillot) [reply Taskfile.yml:62]: @titpetric there's a dependency between the setup step and the `docker` task. Any idea on how can we solve that?

File: ci/tests/tracing/post.sh (hunk 3 above)
Comment (tbuchaillot) [reply ci/tests/tracing/post.sh:3]: It was a reminder for Monday Tomas :D working on it!

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5292/merge
Commit: 6eaa68da5e69349e38c179253f14f31b49a9d615 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5510100449)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5292/merge
Commit: 18c396a8ef7c586d73b7a4459058234174474a26 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5510262124)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5292/merge
Commit: db11b1880a5ebd29a41da0c1f3c16883570d0440 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5510764374)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5292/merge
Commit: b8984fb7ce1a7dfdc24ab798b4bb7261a313df9e 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5523053056)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5292/merge
Commit: 2c3b346d4aa28212d36b4e8408e83df81cfe31e0 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5523173727)

**Review by sredxny — APPROVED** [review]
@tbuchaillot awesome! Excelent job. We need to think how we will introduce tests for goplugins 👀  as it seems like we will need to compile the plugin for each version.

File: Taskfile.yml (hunk 2 above)
Comment (titpetric) [reply Taskfile.yml:54]: @tbuchaillot drop the dependency, but keep the docker target. No suggestions atm

**Pull request merged**